streaming_form_data/_parser.pyx

```diff
@@ -247,62 +252,134 @@ cdef class _Parser:
             elif self.state == ParserState.PS_READING_BODY:
-                buffer_end += 1
 
                 self.delimiter_finder.feed(byte)
                 self.ender_finder.feed(byte)
 
                 if self.delimiter_finder.found():
                     self.state = ParserState.PS_READING_HEADER
 
-                    if buffer_end - buffer_start > self.delimiter_length:
-                        _idx = buffer_end - self.delimiter_length
+                    if idx + 1 < self.delimiter_length:
+                        return 80
+                    match_start = idx + 1 - self.delimiter_length
 
-                        self.on_body(chunk[buffer_start: _idx - 2])
+                    if match_start >= buffer_start:
+                        self.on_body(chunk[buffer_start: match_start])
 
-                        buffer_start = buffer_end = idx + 1
+                        buffer_start = idx + 1
+                    else:
+                        return 90
 
                     self.unset_active_part()
                     self.delimiter_finder.reset()
+
                 elif self.ender_finder.found():
                     self.state = ParserState.PS_END
 
-                    if buffer_end - buffer_start > self.ender_length:
-                        _idx = buffer_end - self.ender_length
+                    if idx + 1 < self.ender_length:
+                        return 100
+                    match_start = idx + 1 - self.ender_length
 
-                        if chunk[_idx - 1] == Constants.LF and \
-                                chunk[_idx - 2] == Constants.CR:
-                            self.on_body(chunk[buffer_start: _idx - 2])
-                        else:
-                            self.on_body(chunk[buffer_start: _idx])
+                    if match_start >= buffer_start:
+                         self.on_body(chunk[buffer_start: match_start])
+                    else:
+                        return 110
 
-                        buffer_start = buffer_end = idx + 1
+                    buffer_start = idx + 1
 
                     self.unset_active_part()
                     self.ender_finder.reset()
+
                 else:
-                    if self.ender_finder.inactive() and \
-                            self.delimiter_finder.inactive() and \
-                            buffer_end - buffer_start > Constants.MaxBufferSize:
-                        _idx = buffer_end - 1
+                    # The following block is for great speed optimization
+                    # The idea is to skip all data not containing
+                    # delimiter starting sequence '\r\n--' when
+                    # we are not already in the middle of potential delimiter
 
-                        self.on_body(chunk[buffer_start: _idx])
+                    if self.delimiter_finder.inactive():
+                        skip_count = self.RewindFastForward(chunk_ptr, idx + 1, chunk_len-1)
```

Comment:
Please rename to `rewind_fast_forward` for consistency with the rest of the codebase.

Reply:
sure. My mistake

Reply:
No worries!